astro/src/diagrams/docs/get-started/core-concepts/integration-points/oauth-grant-with-federation.astro

```diff
@@ -0,0 +1,33 @@
+---
+import Diagram from "src/components/mermaid/SequenceDiagram.astro";
+const { alt } = Astro.props;
+
+//language=Mermaid
+const diagram = `
+sequenceDiagram
+    participant User as User/Browser
+    participant App
+    participant FusionAuth
+    participant IdentityProvider
+
+    User ->> App : View Initial Page<br/>Click Login
+    App ->> User : Redirect To <br/>FusionAuth Authorization URL
+    User ->> FusionAuth : Request Login Page
+    FusionAuth ->> User : Return Login Page
+    User ->> FusionAuth : Clicks On 'Login With Identity Provider'
+    FusionAuth ->> User : Redirect To Identity Provider Authorization URL
+    IdentityProvider ->> IdentityProvider : Validate Credentials
```

Comment:
Before `IdentityProvider: Validate Credentials`, we should have something like
`User ->> IdentityProvider : Enters Credentials`